File: backend/src/main/java/com/prefin/domain/quest/QuestOwned.java
package com.prefin.domain.quest;

import com.prefin.domain.user.Child;
import lombok.*;

import javax.persistence.*;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Builder
public class QuestOwned {
    @Id
    @GeneratedValue
    private Long id;

    private boolean requested;

    private boolean completed;

    private long startDate;

    private long endDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "QUEST_ID")
    private Quest quest;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CHILD_ID")
    private Child child;

    public void updateRequested(boolean requested) {this.requested = requested;}

    public void updateCompleted(boolean completed) {this.completed = completed;}
}

File: mobile/Prefin/app/src/main/java/com/prefin/model/dto/QuestOwned.kt
package com.prefin.model.dto

data class QuestOwned(
    var id: Long = 0L,
    var requested: Boolean = false,
    var completed: Boolean = false,
    var startDate: Long = 0L,
    var endDate: Long = 0L,
    var quest: Quest = Quest(),
)
